## Final Project Submission

Please fill out:
* Student name: Merlah Mendoza Andang'o 
* Student pace: Full time
* Scheduled project review date/time: 
* Instructor name: Mark Tiba
* Blog post URL:


# 1. Introduction

## 1.1 Overview



The main goal of the project is to come up with insights to help Microsoft start and excel in the movie business.We will use exploratory data analysis to come up with actionable insights to help decide what type of films Microsoft should create.

## 1.2 Objectives

### 1.2.1 General objectives



Explore and analyze the types of films that are currently doing the best at the box office and using the findings to help decide the types of films to create


### 1.2.2 Specific objectives

i. Loading datasets into pandas <br>
ii. Retrieving data from a DataFrame <br>
iii. Data preparation annd cleaning <br>
iv. Exploratory analysis <br>
v. Data visualization <br>

## 1.3 Questions to answer

1. What are the highly rated genres?
2. Does language affect popularity of a movie?
3. Does rating affect movie revenue?
4. 
What type of films are currently doing the best at the box office?<br>
 a) What films are the highest grossing in the box office?
b) What are the highest rated movies in the box office?
c)What are the genres of the highest grosssing and highest rated movies in the box office?
d) Does rating affect movie revenue?
e) Does budget affect movie profits?
f) Is there a best time to debut a movie?

# 2. Data Munging

## 2.1 The Data

The datasets are stored in the 'zippedData' folder which were sourced from: <br>
- Box Office MojoLinks
    - bom.movie_gross.csv
- IMDBLinks
    - im.db
- Rotten TomatoesLinks
    - rt.movie_info.tsv
- TheMovieDBLinks
    - tmdb.movies.csv
- The Numbers
    - tn.movie_budgets.csv

## 2.2 Importing the relevant libraries
Begin by importing pandas, numpy, csv with their aliases

In [54]:
# Your code here - remember to use markdown cells for comments as well!
#importing libraries
import pandas as pd
import sqlite3
import csv
import numpy as np
from datetime import datetime


### Loading the datasets


movies_budget = pd.read_csv('zippedData/tn.movie_budgets.csv') <br>
movies_gross = pd.read_csv('zippedData/bom.movie_gross.csv') <br>
tmd = pd.read_csv('zippedData/tmdb.movies.csv',index_col=0) <br>
reviews = pd.read_csv('zippedData/rt.reviews.tsv', delimiter='\t',encoding='unicode_escape') <br>
conn = sqlite3.connect('zippedData/im.db') <br>
movie_info = pd.read_csv('zippedData/rt.movie_info.tsv', delimiter ='\t', encoding='unicode_escape' )


## 2.3 Exploring the datasets


### 2.3.1 The Numbers dataset
The Numbers dataset gives us information on the budget and returns. <br>


In [55]:



#Load the data from tn.movie_budgets.csv into a dataframe
movies_budget = pd.read_csv('zippedData/tn.movie_budgets.csv')

#Cast to numeric data type
movies_budget['production_budget'] = movies_budget['production_budget'].replace('[^.0-9]', '', regex=True).astype('float32')
movies_budget['domestic_gross'] = movies_budget['domestic_gross'].replace('[^.0-9]', '', regex=True).astype('float32')
movies_budget['worldwide_gross'] = movies_budget['worldwide_gross'].replace('[^.0-9]', '',regex = True).astype('float64')
movies_budget.head(5)

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  
0        425000000.0     760507648.0     2.776345e+09  
1        410600000.0     241063872.0     1.045664e+09  
2        350000000.0      42762352.0     1.497624e+08  
3        330600000.0     459005856.0     1.403014e+09  
4        317000000.0     620181376.0     1.316722e+09

The production budget, domestic gross, and worldwide gross are then cast to float type for mathematical manipulation.

In [56]:
movies_budget.shape

(5782, 6)

In [57]:
movies_budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5782 non-null   int64  
 1   release_date       5782 non-null   object 
 2   movie              5782 non-null   object 
 3   production_budget  5782 non-null   float32
 4   domestic_gross     5782 non-null   float32
 5   worldwide_gross    5782 non-null   float64
dtypes: float32(2), float64(1), int64(1), object(2)
memory usage: 226.0+ KB


In [58]:
movies_budget.isna().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

The movie budget Data frame has 5,782 rows and 6 columns namely: id,release_date, movie, production_budget, domestic gross and worldwide_gross. The data types are as float,object and int. The are no null values.

Finding out the movies budget, first we get the international gross by subtracting the domestic gross from the woorldwide gross

In [59]:
#getting the international gross
movies_budget['international_gross'] = movies_budget['worldwide_gross'] - movies_budget['domestic_gross']
cols = list(movies_budget.columns)

movies_budget.head(10)

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
5   6  Dec 18, 2015         Star Wars Ep. VII: The Force Awakens   
6   7  Apr 27, 2018                       Avengers: Infinity War   
7   8  May 24, 2007   Pirates of the Caribbean: At World’s End   
8   9  Nov 17, 2017                               Justice League   
9  10   Nov 6, 2015                                      Spectre   

   production_budget  domestic_gross  worldwide_gross  international_gross  
0        425000000.0     760507648.0     2.776345e+09         2.015838e+09  
1        410600000.0     241063872.0     1.045664e+09         8.046000e+08  
2        350000000.0      42762352.0     1.497624e+08         1.070000e+08  
3        330600000.0     459005856.0     1.403014e+09         9.440081e+08  
4        317000000.0     620181376.0     1.316722e+09         6.965404e+08  
5        306000000.0     936662208.0     2.053311e+09         1.116649e+09  
6        300000000.0     678815488.0     2.048134e+09         1.369319e+09  
7        300000000.0     309420416.0     9.634204e+08         6.540000e+08  
8        300000000.0     229024288.0     6.559452e+08         4.269209e+08  
9        300000000.0     200074176.0     8.796209e+08         6.795467e+08

The release date is an object, in order to filter the years convert to date time type.

In [60]:
#converting object to dateTime
movies_budget['release_date'] = pd.to_datetime(movies_budget['release_date'], errors='coerce')
movies_budget['year'] = movies_budget['release_date'].dt.strftime('%Y')
movies_budget.head(10)

id release_date                                        movie  \
0   1   2009-12-18                                       Avatar   
1   2   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2   3   2019-06-07                                 Dark Phoenix   
3   4   2015-05-01                      Avengers: Age of Ultron   
4   5   2017-12-15            Star Wars Ep. VIII: The Last Jedi   
5   6   2015-12-18         Star Wars Ep. VII: The Force Awakens   
6   7   2018-04-27                       Avengers: Infinity War   
7   8   2007-05-24   Pirates of the Caribbean: At World’s End   
8   9   2017-11-17                               Justice League   
9  10   2015-11-06                                      Spectre   

   production_budget  domestic_gross  worldwide_gross  international_gross  \
0        425000000.0     760507648.0     2.776345e+09         2.015838e+09   
1        410600000.0     241063872.0     1.045664e+09         8.046000e+08   
2        350000000.0      42762352.0     1.497624e+08         1.070000e+08   
3        330600000.0     459005856.0     1.403014e+09         9.440081e+08   
4        317000000.0     620181376.0     1.316722e+09         6.965404e+08   
5        306000000.0     936662208.0     2.053311e+09         1.116649e+09   
6        300000000.0     678815488.0     2.048134e+09         1.369319e+09   
7        300000000.0     309420416.0     9.634204e+08         6.540000e+08   
8        300000000.0     229024288.0     6.559452e+08         4.269209e+08   
9        300000000.0     200074176.0     8.796209e+08         6.795467e+08   

   year  
0  2009  
1  2011  
2  2019  
3  2015  
4  2017  
5  2015  
6  2018  
7  2007  
8  2017  
9  2015

In [61]:
movies_budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 8 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   id                   5782 non-null   int64         
 1   release_date         5782 non-null   datetime64[ns]
 2   movie                5782 non-null   object        
 3   production_budget    5782 non-null   float32       
 4   domestic_gross       5782 non-null   float32       
 5   worldwide_gross      5782 non-null   float64       
 6   international_gross  5782 non-null   float64       
 7   year                 5782 non-null   object        
dtypes: datetime64[ns](1), float32(2), float64(2), int64(1), object(2)
memory usage: 316.3+ KB


#### Filtering the Data
Here we will drop the rows without the values for the domestic, international and worldwide gross.

In [62]:
movies_budget.sort_values('worldwide_gross')

id release_date                                    movie  \
5037  38   2019-04-23  Living Dark: The Story of Ted the Caver   
3975  76   2015-05-15                           Pound of Flesh   
4627  28   2011-06-28                                     2:13   
4628  29   2013-01-29  Batman: The Dark Knight Returns, Part 2   
3947  48   2019-06-21                           Burn Your Maps   
...   ..          ...                                      ...   
33    34   2015-06-12                           Jurassic World   
6      7   2018-04-27                   Avengers: Infinity War   
5      6   2015-12-18     Star Wars Ep. VII: The Force Awakens   
42    43   1997-12-19                                  Titanic   
0      1   2009-12-18                                   Avatar   

      production_budget  domestic_gross  worldwide_gross  international_gross  \
5037          1750000.0             0.0     0.000000e+00         0.000000e+00   
3975          7500000.0             0.0     0.000000e+00         0.000000e+00   
4627          3500000.0             0.0     0.000000e+00         0.000000e+00   
4628          3500000.0             0.0     0.000000e+00         0.000000e+00   
3947          8000000.0             0.0     0.000000e+00         0.000000e+00   
...                 ...             ...              ...                  ...   
33          215000000.0     652270656.0     1.648855e+09         9.965842e+08   
6           300000000.0     678815488.0     2.048134e+09         1.369319e+09   
5           306000000.0     936662208.0     2.053311e+09         1.116649e+09   
42          200000000.0     659363968.0     2.208208e+09         1.548844e+09   
0           425000000.0     760507648.0     2.776345e+09         2.015838e+09   

      year  
5037  2019  
3975  2015  
4627  2011  
4628  2013  
3947  2019  
...    ...  
33    2015  
6     2018  
5     2015  
42    1997  
0     2009  

[5782 rows x 8 columns]

In [63]:
#drop all all rows without the gross values
movies_budget_drop = movies_budget[(movies_budget['domestic_gross'] == 0) & (movies_budget['worldwide_gross'] == 0)]

In [64]:
movies_budget.drop(movies_budget_drop.index, axis=0, inplace=True)
movies_budget.head(5)

id release_date                                        movie  \
0   1   2009-12-18                                       Avatar   
1   2   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2   3   2019-06-07                                 Dark Phoenix   
3   4   2015-05-01                      Avengers: Age of Ultron   
4   5   2017-12-15            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  international_gross  \
0        425000000.0     760507648.0     2.776345e+09         2.015838e+09   
1        410600000.0     241063872.0     1.045664e+09         8.046000e+08   
2        350000000.0      42762352.0     1.497624e+08         1.070000e+08   
3        330600000.0     459005856.0     1.403014e+09         9.440081e+08   
4        317000000.0     620181376.0     1.316722e+09         6.965404e+08   

   year  
0  2009  
1  2011  
2  2019  
3  2015  
4  2017

In order to have relevant budgeting information, filter the years to get only the data from 2011

In [65]:
#convert from object to string
movies_budget['year'] = movies_budget['year'].astype(int)


In [66]:
#drop rows which are not between the years 2015 and 2022
drop_=  movies_budget[(movies_budget['year'] <= 2014)]
movies_budget.drop(drop_.index, axis=0, inplace=True)
movies_budget.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 797 entries, 2 to 5748
Data columns (total 8 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   id                   797 non-null    int64         
 1   release_date         797 non-null    datetime64[ns]
 2   movie                797 non-null    object        
 3   production_budget    797 non-null    float32       
 4   domestic_gross       797 non-null    float32       
 5   worldwide_gross      797 non-null    float64       
 6   international_gross  797 non-null    float64       
 7   year                 797 non-null    int32         
dtypes: datetime64[ns](1), float32(2), float64(2), int32(1), int64(1), object(1)
memory usage: 46.7+ KB


In [67]:
movies_budget['year'].value_counts()

2015    242
2016    204
2017    162
2018    140
2019     49
Name: year, dtype: int64

In [68]:
movies_budget.describe()

id  production_budget  domestic_gross  worldwide_gross  \
count  797.000000              797.0           797.0     7.970000e+02   
mean    50.634881         44454288.0      57354036.0     1.477667e+08   
std     29.339221         57002492.0      98196584.0     2.607258e+08   
min      1.000000            25000.0             0.0     2.600000e+01   
25%     25.000000          8000000.0       3000342.0     8.200000e+06   
50%     50.000000         22000000.0      24431472.0     4.517374e+07   
75%     78.000000         55000000.0      63859436.0     1.594876e+08   
max    100.000000        350000000.0     936662208.0     2.053311e+09   

       international_gross         year  
count         7.970000e+02   797.000000  
mean          9.041263e+07  2016.435383  
std           1.710463e+08     1.255101  
min          -1.000000e+00  2015.000000  
25%           2.053330e+06  2015.000000  
50%           1.928622e+07  2016.000000  
75%           9.237100e+07  2017.000000  
max           1.369319e+09  2019.000000

Finding the percentage profits of the movies based on the production budget and the world wide gross.

In [69]:
movies_budget['profit/loss'] = (movies_budget['worldwide_gross'] - movies_budget['production_budget'])
movies_budget['profit/loss'] = movies_budget['profit/loss'].astype(int)      
movies_budget['profit/loss'].tail()

5701    -66201
5719    -41626
5737    114822
5744    -23453
5748    464792
Name: profit/loss, dtype: int32

In [70]:
movies_budget.shape

(797, 9)

In [71]:
movies_budget

id release_date                                 movie  \
2      3   2019-06-07                          Dark Phoenix   
3      4   2015-05-01               Avengers: Age of Ultron   
4      5   2017-12-15     Star Wars Ep. VIII: The Last Jedi   
5      6   2015-12-18  Star Wars Ep. VII: The Force Awakens   
6      7   2018-04-27                Avengers: Infinity War   
...   ..          ...                                   ...   
5701   2   2015-07-17             Dawn of the Crescent Moon   
5719  20   2015-07-31                              Counting   
5737  38   2016-03-18                                Krisha   
5744  45   2017-01-27                                 Emily   
5748  49   2015-09-01                                Exeter   

      production_budget  domestic_gross  worldwide_gross  international_gross  \
2           350000000.0      42762352.0     1.497624e+08         1.070000e+08   
3           330600000.0     459005856.0     1.403014e+09         9.440081e+08   
4           317000000.0     620181376.0     1.316722e+09         6.965404e+08   
5           306000000.0     936662208.0     2.053311e+09         1.116649e+09   
6           300000000.0     678815488.0     2.048134e+09         1.369319e+09   
...                 ...             ...              ...                  ...   
5701            75000.0          8799.0     8.799000e+03         0.000000e+00   
5719            50000.0          8374.0     8.374000e+03         0.000000e+00   
5737            30000.0        144822.0     1.448220e+05         0.000000e+00   
5744            27000.0          3547.0     3.547000e+03         0.000000e+00   
5748            25000.0             0.0     4.897920e+05         4.897920e+05   

      year  profit/loss  
2     2019   -200237650  
3     2015   1072413963  
4     2017    999721747  
5     2015   1747311220  
6     2018   1748134200  
...    ...          ...  
5701  2015       -66201  
5719  2015       -41626  
5737  2016       114822  
5744  2017       -23453  
5748  2015       464792  

[797 rows x 9 columns]

### 2.3.2 Box Office Mojo Dataset

The bom dataset gives information on the domestic gross values and the foreign gross.

In [72]:
#previewing the second dataset
movies_gross = pd.read_csv('zippedData/bom.movie_gross.csv')
movies_gross.tail(10)

title      studio  domestic_gross foreign_gross  \
3377               Let Yourself Go        Men.         12200.0           NaN   
3378                 Hannah (2018)         PDF         11700.0           NaN   
3379                      Souvenir      Strand         11400.0           NaN   
3380  Furious (Legend of Kolovrat)     CARUSEL         10000.0           NaN   
3381           Beauty and the Dogs       Osci.          8900.0           NaN   
3382                     The Quake       Magn.          6200.0           NaN   
3383   Edward II (2018 re-release)          FM          4800.0           NaN   
3384                      El Pacto        Sony          2500.0           NaN   
3385                      The Swan  Synergetic          2400.0           NaN   
3386             An Actor Prepares       Grav.          1700.0           NaN   

      year  
3377  2018  
3378  2018  
3379  2018  
3380  2018  
3381  2018  
3382  2018  
3383  2018  
3384  2018  
3385  2018  
3386  2018

In [73]:
#find out the shape of the dataset
movies_gross.shape

(3387, 5)

In [74]:
movies_gross['year'].unique()

array([2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018], dtype=int64)

In [75]:
movies_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [76]:
movies_gross.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [77]:
#drop the rows without gross values
movies_gross.dropna(subset=['foreign_gross','domestic_gross'], inplace=True)
movies_gross.isna().sum()

title             0
studio            2
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

#### 2.3.3 TheMovieDB dataset

The tmdb.movies dataset gives us information on the popularity of a movie with respect to the language.

In [78]:
tmd = pd.read_csv('zippedData/tmdb.movies.csv',index_col=0)
tmd.head(5)

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [79]:
tmd['original_language'].value_counts().head(5)

en    23291
fr      507
es      455
ru      298
ja      265
Name: original_language, dtype: int64

In [80]:
tmd.shape

(26517, 9)

The dataset has 26,517 rows and 9 columns

In [81]:
tmd.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


In [82]:
tmd.duplicated().sum()

1020

In [83]:
#drop all duplicate rows based on the titles columns
tmd = tmd.drop_duplicates(subset = ['title','original_title'], keep = 'first')
tmd.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 24842 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          24842 non-null  object 
 1   id                 24842 non-null  int64  
 2   original_language  24842 non-null  object 
 3   original_title     24842 non-null  object 
 4   popularity         24842 non-null  float64
 5   release_date       24842 non-null  object 
 6   title              24842 non-null  object 
 7   vote_average       24842 non-null  float64
 8   vote_count         24842 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 1.9+ MB


In [84]:
tmd.duplicated().sum()

0

The data has no duplicates 

In [85]:
tmd['popularity'].max()


80.773

### 2.3.4 Rotten tomatoes dataset

In [86]:
reviews = pd.read_csv('zippedData/rt.reviews.tsv', delimiter='\t',encoding='unicode_escape')
reviews.tail(10)

id                                             review rating   fresh  \
54422  2000  Arguably the best script that Besson has writt...  3.5/5   fresh   
54423  2000          Despite the title, Wasabi isn't so spicy.      C  rotten   
54424  2000  Dawdles and drags when it should pop; it doesn...  1.5/5  rotten   
54425  2000  Despite Besson's high-profile name being Wasab...    NaN   fresh   
54426  2000  The film lapses too often into sugary sentimen...    NaN  rotten   
54427  2000  The real charm of this trifle is the deadpan c...    NaN   fresh   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic           publisher                date  
54422         Wade Major           0  Boxoffice Magazine  September 27, 2002  
54423                NaN           0           E! Online  September 27, 2002  
54424     Manohla Dargis           1   Los Angeles Times  September 26, 2002  
54425         Andy Klein           0           New Times  September 26, 2002  
54426       Paul Malcolm           1         L.A. Weekly  September 26, 2002  
54427      Laura Sinagra           1       Village Voice  September 24, 2002  
54428  Michael Szymanski           0          Zap2it.com  September 21, 2005  
54429       Emanuel Levy           0     EmanuelLevy.Com       July 17, 2005  
54430   Christopher Null           0      Filmcritic.com   September 7, 2003  
54431    Nicolas Lacroix           0        Showbizz.net   November 12, 2002

In [87]:
reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


In [88]:
reviews.isna().sum()

id                0
review         5563
rating        13517
fresh             0
critic         2722
top_critic        0
publisher       309
date              0
dtype: int64

In [89]:
reviews.duplicated().sum()

9

In [90]:
reviews.dropna(axis=0, inplace = True)
reviews.head(10)

id                                             review rating   fresh  \
0    3  A distinctly gallows take on contemporary fina...    3/5   fresh   
6    3  Quickly grows repetitive and tiresome, meander...      C  rotten   
7    3  Cronenberg is not a director to be daunted by ...    2/5  rotten   
11   3  While not one of Cronenberg's stronger films, ...     B-   fresh   
12   3  Robert Pattinson works mighty hard to make Cos...    2/4  rotten   
13   3  The anger over the injustice of the financial ...      B   fresh   
14   3  For those who like their Cronenberg thick and ...    3/4   fresh   
15   3  For better or worse - often both - Cosmopolis ...    3/5   fresh   
16   3  For one of the smartest films I've seen in a w...    4/5   fresh   
17   3      Its major problem is that it's not cinematic.      B   fresh   

             critic  top_critic                  publisher                date  
0        PJ Nabarro           0            Patrick Nabarro   November 10, 2018  
6    Eric D. Snider           0            EricDSnider.com       July 17, 2013  
7      Matt Kelemen           0         Las Vegas CityLife      April 21, 2013  
11     Emanuel Levy           0            EmanuelLevy.Com    February 3, 2013  
12   Christian Toto           0              Big Hollywood    January 15, 2013  
13     Robert Roten           0        Laramie Movie Scope     January 7, 2013  
14      Marty Mapes           0                Movie Habit    October 20, 2012  
15        Adam Ross           0             The Aristocrat  September 27, 2012  
16    Patrick Kolan           0             Shotgun Cinema  September 26, 2012  
17  Dennis Schwartz           0  Ozus' World Movie Reviews  September 25, 2012

From the reviews dataframe it is impossible to get valid data that can be used to draw conclusion on the ratings and the reviews

### 2.3.4 IMDB dataset

In [91]:
conn = sqlite3.connect('zippedData/im.db')
cur = conn.cursor()

#### Find the table names in im.db

In [92]:
cur.execute("""SELECT name FROM sqlite_master WHERE type = 'table';""")
# Fetch the result and store it in table_names
table_names = cur.fetchall()
table_names

[('movie_basics',),
 ('directors',),
 ('known_for',),
 ('movie_akas',),
 ('movie_ratings',),
 ('persons',),
 ('principals',),
 ('writers',)]

Im.db has 8 tables. The movie_ratings and movie_basics are more relevant to our study as they will help gain information on the highly rated genres.

In [93]:
ratings = pd.read_sql_query("""
SELECT movie_id, start_year, runtime_minutes, genres, averagerating, numvotes
FROM movie_basics
JOIN movie_ratings
    USING(movie_id)
WHERE start_year >= 201
LIMIT 10
;""", conn)

ratings

movie_id  start_year  runtime_minutes                      genres  \
0  tt0063540        2013            175.0          Action,Crime,Drama   
1  tt0066787        2019            114.0             Biography,Drama   
2  tt0069049        2018            122.0                       Drama   
3  tt0069204        2018              NaN                Comedy,Drama   
4  tt0100275        2017             80.0        Comedy,Drama,Fantasy   
5  tt0112502        2017              NaN             Horror,Thriller   
6  tt0137204        2017             83.0  Adventure,Animation,Comedy   
7  tt0146592        2010            136.0                       Drama   
8  tt0154039        2010            100.0                     History   
9  tt0159369        2013            180.0                 Documentary   

   averagerating  numvotes  
0            7.0        77  
1            7.2        43  
2            6.9      4517  
3            6.1        13  
4            6.5       119  
5            4.1        32  
6            8.1       263  
7            6.8       451  
8            4.6        64  
9            7.6        53

In [94]:
movie_akas = pd.read_sql_query("""SELECT * 
FROM movie_akas
;""",conn)

movie_akas.tail(10)

movie_id  ordering                          title region language  \
331693  tt9726638         3  Qi Tian Da Sheng Huo Yan Shan     CN      yue   
331694  tt9755806         1                      Big Shark     US     None   
331695  tt9755806         2                  Большая Акула     RU     None   
331696  tt9755806         3                      Big Shark   None     None   
331697  tt9827784         1             Sayonara kuchibiru     JP     None   
331698  tt9827784         2             Sayonara kuchibiru   None     None   
331699  tt9827784         3                  Farewell Song    XWW       en   
331700  tt9880178         1                    La atención   None     None   
331701  tt9880178         2                    La atención     ES     None   
331702  tt9880178         3                  The Attention    XWW       en   

              types attributes  is_original_title  
331693  imdbDisplay       None                0.0  
331694         None       None                0.0  
331695         None       None                0.0  
331696     original       None                1.0  
331697         None       None                0.0  
331698     original       None                1.0  
331699  imdbDisplay       None                0.0  
331700     original       None                1.0  
331701         None       None                0.0  
331702  imdbDisplay       None                0.0

In [95]:
#find the languages 
movie_akas['language'].unique()
movie_akas['language'].value_counts().head(10)

en     22895
tr      3847
bg      3609
fr      3576
he      2680
sv       965
cmn      727
fa       482
hi       307
yue      286
Name: language, dtype: int64

In [96]:
movie_info = pd.read_csv('zippedData/rt.movie_info.tsv', delimiter ='\t', encoding='unicode_escape' )
movie_info

id                                           synopsis rating  \
0        1  This gritty, fast-paced, and innovative police...      R   
1        3  New York City, not-too-distant-future: Eric Pa...      R   
2        5  Illeana Douglas delivers a superb performance ...      R   
3        6  Michael Douglas runs afoul of a treacherous su...      R   
4        7                                                NaN     NR   
...    ...                                                ...    ...   
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
0                   Action and Adventure|Classics|Drama    William Friedkin   
1                     Drama|Science Fiction and Fantasy    David Cronenberg   
2                     Drama|Musical and Performing Arts      Allison Anders   
3                            Drama|Mystery and Suspense      Barry Levinson   
4                                         Drama|Romance      Rodney Bennett   
...                                                 ...                 ...   
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
0                                       Ernest Tidyman   Oct 9, 1971   
1                         David Cronenberg|Don DeLillo  Aug 17, 2012   
2                                       Allison Anders  Sep 13, 1996   
3                      Paul Attanasio|Michael Crichton   Dec 9, 1994   
4                                         Giles Cooper           NaN   
...                                                ...           ...   
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
0     Sep 25, 2001      NaN         NaN  104 minutes                NaN  
1      Jan 1, 2013        $     600,000  108 minutes  Entertainment One  
2     Apr 18, 2000      NaN         NaN  116 minutes                NaN  
3     Aug 27, 1997      NaN         NaN  128 minutes                NaN  
4              NaN      NaN         NaN  200 minutes                NaN  
...            ...      ...         ...          ...                ...  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures  

[1560 rows x 12 columns]

In [97]:
movie_info.isna().sum()

id                 0
synopsis          62
rating             3
genre              8
director         199
writer           449
theater_date     359
dvd_date         359
currency        1220
box_office      1220
runtime           30
studio          1066
dtype: int64

In [98]:
movie_info.dropna(inplace = True)

In [99]:
movie_info.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 235 entries, 1 to 1545
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            235 non-null    int64 
 1   synopsis      235 non-null    object
 2   rating        235 non-null    object
 3   genre         235 non-null    object
 4   director      235 non-null    object
 5   writer        235 non-null    object
 6   theater_date  235 non-null    object
 7   dvd_date      235 non-null    object
 8   currency      235 non-null    object
 9   box_office    235 non-null    object
 10  runtime       235 non-null    object
 11  studio        235 non-null    object
dtypes: int64(1), object(11)
memory usage: 23.9+ KB


In [100]:
movie_info.head(10)

id                                           synopsis rating  \
1    3  New York City, not-too-distant-future: Eric Pa...      R   
6   10  Some cast and crew from NBC's highly acclaimed...  PG-13   
7   13  Stewart Kane, an Irishman living in the Austra...      R   
15  22  Two-time Academy Award Winner Kevin Spacey giv...      R   
18  25  From ancient Japan's most enduring tale, the e...  PG-13   
19  26  A comic series of short vignettes build on one...      R   
29  41  Ken Loach directs Sweet Sixteen, a gritty dram...      R   
48  61  Death once again shows he's determined to get ...      R   
61  77  Until he was downsized, affable, amiable Larry...  PG-13   
69  87  "Mao's Last Dancer" is the inspiring true stor...     PG   

                                                genre             director  \
1                   Drama|Science Fiction and Fantasy     David Cronenberg   
6                                              Comedy          Jake Kasdan   
7                                               Drama         Ray Lawrence   
15                  Comedy|Drama|Mystery and Suspense  George Hickenlooper   
18  Action and Adventure|Drama|Science Fiction and...     Carl Erik Rinsch   
19  Art House and International|Comedy|Drama|Music...         Jim Jarmusch   
29                                              Drama            Ken Loach   
48  Action and Adventure|Drama|Horror|Mystery and ...           James Wong   
61                                       Comedy|Drama            Tom Hanks   
69  Art House and International|Drama|Musical and ...      Bruce Beresford   

                              writer  theater_date      dvd_date currency  \
1       David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
6                         Mike White  Jan 11, 2002  Jun 18, 2002        $   
7   Raymond Carver|Beatrix Christian  Apr 27, 2006   Oct 2, 2007        $   
15                     Norman Snider  Dec 17, 2010   Apr 5, 2011        $   
18        Chris Morgan|Hossein Amini  Dec 25, 2013   Apr 1, 2014        $   
19                      Jim Jarmusch  May 14, 2004  Sep 21, 2004        $   
29                      Paul Laverty  May 16, 2003   Oct 7, 2003        $   
48            James Wong|Glen Morgan  Feb 10, 2006  Jul 25, 2006        $   
61            Tom Hanks|Nia Vardalos   Jul 1, 2011  Nov 15, 2011        $   
69                         Jan Sardi  Aug 20, 2010   May 3, 2011        $   

    box_office      runtime                  studio  
1      600,000  108 minutes       Entertainment One  
6   41,032,915   82 minutes      Paramount Pictures  
7      224,114  123 minutes  Sony Pictures Classics  
15   1,039,869  108 minutes            ATO Pictures  
18  20,518,224  127 minutes      Universal Pictures  
19   1,971,135   96 minutes                     MGM  
29     201,010  106 minutes        Lions Gate Films  
48  54,100,000   93 minutes         New Line Cinema  
61  35,565,975   99 minutes      Universal Pictures  
69   4,806,750  117 minutes    Samuel Goldwyn Films

### 3. Merging Datasets

In [101]:
#renaming the column for merging

movies_gross.rename(columns={'title':'movie'}, inplace=True)

In [102]:
tmd.rename(columns={'original_title':'movie'}, inplace=True)

In [103]:
#merging the movie_budget dataframe and movie_gross 

df_budget_and_gross = movies_budget.merge(movies_gross, how = 'inner', on ='movie')

In [104]:
df_budget_and_gross.shape

(409, 13)

Merging tmd dataframe with the movie_akas on language

In [105]:
language_ = tmd.merge(movie_akas, how = 'inner', on ='title')

In [106]:
language_.shape

(39753, 16)

Merge BoM with 

In [107]:
df_rating_genres = df_budget_and_gross.merge(tmd, how = 'inner', on = 'movie')

In [109]:
studio_info = movie_info.merge(movies_gross, how ='inner', on ='studio')

In [111]:
final_df = pd.merge(df_rating_genres,studio_info, how = 'inner', on ='movie')
final_df.head(5)

id_x release_date_x                                        movie  \
0     9     2017-11-17                               Justice League   
1    18     2016-03-25           Batman v Superman: Dawn of Justice   
2    60     2018-11-16  Fantastic Beasts: The Crimes of Grindelwald   
3    78     2017-03-10                           Kong: Skull Island   
4    80     2017-10-06                            Blade Runner 2049   

   production_budget  domestic_gross_x  worldwide_gross  international_gross  \
0        300000000.0       229024288.0      655945209.0          426920921.0   
1        250000000.0       330360192.0      867500281.0          537140089.0   
2        200000000.0       159555904.0      652220086.0          492664182.0   
3        185000000.0       168052816.0      561072059.0          393019243.0   
4        185000000.0        92054160.0      259357408.0          167303248.0   

   year_x  profit/loss studio_x  ...           writer theater_date  \
0    2017    355945209       WB  ...  Brian Helgeland  Oct 8, 2003   
1    2016    617500281       WB  ...  Brian Helgeland  Oct 8, 2003   
2    2018    452220086       WB  ...  Brian Helgeland  Oct 8, 2003   
3    2017    376072059       WB  ...  Brian Helgeland  Oct 8, 2003   
4    2017     74357408       WB  ...  Brian Helgeland  Oct 8, 2003   

      dvd_date currency  box_office      runtime  studio_y domestic_gross  \
0  Jun 8, 2004        $  88,800,000  137 minutes        WB    229000000.0   
1  Jun 8, 2004        $  88,800,000  137 minutes        WB    330400000.0   
2  Jun 8, 2004        $  88,800,000  137 minutes        WB    159600000.0   
3  Jun 8, 2004        $  88,800,000  137 minutes        WB    168100000.0   
4  Jun 8, 2004        $  88,800,000  137 minutes        WB     92100000.0   

  foreign_gross_y  year  
0       428900000  2017  
1       543300000  2016  
2       494100000  2018  
3       398600000  2017  
4       167200000  2017  

[5 rows x 36 columns]

In [ ]:
#Dropping the unecessary columns
final_df = 

In [ ]:
# rounding up the production budget,domestic and worldwide gross to millions

df_final['production_budget'] = df_final['production_budget']/1000000
df_final['domestic_gross'] = df_final['domestic_gross']/1000000
df_final['worldwide_gross'] = df_final['worldwide_gross']/1000000

In [ ]:
# renaming the columns

df_final.rename(columns={'release_date_x':'release_date','primary_name':'directors_name','production_budget':'production_budget_millions','domestic_gros

In [ ]:
column_names ='release_date','movie','production_budget_millions','domestic_gross_millions','worldwide_gross_millions','movie_profits_millions','ge

In [ ]:
# reordering the columns

df_final.reindex(columns=column_names)

In [ ]:
# checking for null values

df_final.isna().sum()

In [ ]:
# checking for duplicates

df_final.duplicated(['movie','release_date','start_year']).sum()

In [ ]:
df_final.drop_duplicates(subset= ['movie','release_date'] ,inplace=True)

In [ ]:
# Exporting this combined dataframe to a csv

df_final.to_csv('Final Combined Microsoft Data.csv')

# 3. Exploratory Data Analysis (EDA)

In [ ]:
# sorting to find movies with the highest ratings per votes cast

movie_ratings = df_final.sort_values(by = ['vote_count', 'vote_average'],ascending=(False, False)).head(10)
movie_ratings = movie_ratings.sort_values('vote_average',ascending=False).reset_index(drop=True)

In [ ]:
# movies with the highest ratings

movie_ratings.loc[:,['movie','vote_average','directors_name']]

In [ ]:
# sorting values to find movies with the highest grossing

df_biggest_gross = df_final.sort_values('worldwide_gross_millions', ascending = False).head(10)
df_biggest_gross.reset_index(drop=True)